src/client/WebhookClient.js

```diff
@@ -10,19 +11,51 @@ const Webhook = require('../structures/Webhook');
  */
 class WebhookClient extends BaseClient {
   /**
-   * @param {Snowflake} id The webhook's id
-   * @param {string} token Token of the webhook
+   * The data for the webhook client containing either an id and token or just a URL
+   * @typedef {Object} WebhookClientData
+   * @property {(WebhookClientDataIdWithToken|WebhookClientDataURL)} data The data for the webhook client
+   */
+
+  /**
+   * The data for the webhook client containing both an id and token
+   * @typedef {Object} WebhookClientDataIdWithToken
+   * @property {Snowflake} id The webhook's id
+   * @property {string} token Token of the webhook
+   */
+
+  /**
+   * The data for the webhook client containing a URL
+   * @typedef {Object} WebhookClientDataURL
+   * @property {string} url The full URL of the webhook
+   */
+
+  /**
+   * @param {WebhookClientData} data The data of the webhook
    * @param {ClientOptions} [options] Options for the client
    * @example
    * // Create a new webhook and send a message
-   * const hook = new Discord.WebhookClient('1234', 'abcdef');
+   * const hook = new Discord.WebhookClient({ id: '1234', token: 'abcdef' });
    * hook.send('This will send a message').catch(console.error);
    */
-  constructor(id, token, options) {
+  constructor(data, options) {
     super(options);
     Object.defineProperty(this, 'client', { value: this });
-    this.id = id;
-    Object.defineProperty(this, 'token', { value: token, writable: true, configurable: true });
+    let id, token;
+
+    if ('url' in data) {
+      const url = data.url.match(
+        // eslint-disable-next-line no-useless-escape
+        /^https?:\/\/(?:canary|ptb)?\.?(?:discord|discordapp)\.com\/api\/webhooks(?:\/v[0-9]\d*)?\/([^\/]+)\/([^\/]+)/i,
```

Comment:
I'm not sure if we should be allowing domains with discordapp.com still? Webhooks links generated now across the release channels won't be using "discordapp", and I feel if people are still using that, they should update their webhook url instead?

Reply:
Yeah we can probably remove the "app" part.